### PR TITLE
[6571] Remove logical resource from WorkflowCancellationDialog, update component for it to show empty state if no items

### PR DESCRIPTION
**Pull request by jvega190**
- Remove logical resource from WorkflowCancellationDialog, update component for it to show empty state if no items

https://github.com/craftercms/craftercms/issues/6571

### CODE REVIEW

File: ui/app/src/components/WorkflowCancellationDialog/WorkflowCancellationDialogContainer.tsx [hunk 1]
```diff
@@ -37,48 +36,30 @@ const useStyles = makeStyles()((theme) => ({
 }));
 
 export function WorkflowCancellationDialogContainer(props: WorkflowCancellationDialogContainerProps) {
-  const { items, onClose, onContinue } = props;
+  const { items = [], onClose, onContinue } = props;
```
Comment (rart) [inline ui/app/src/components/WorkflowCancellationDialog/WorkflowCancellationDialogContainer.tsx]: Remove default and if items is nullish, render an empty state with "No items were provided as input"

File: ui/app/src/components/WorkflowCancellationDialog/utils.ts [hunk 2]
```diff
@@ -25,7 +24,7 @@ export type Source = SandboxItem[];
 export type Return = Omit<Source, 'error'>;
 
 export interface WorkflowCancellationDialogUIProps {
-  resource: Resource<Return>;
+  items: Return;
```
Comment (rart) [inline ui/app/src/components/WorkflowCancellationDialog/utils.ts]:
```suggestion
  items: SandboxItem[];
```

Is this `Return` used elsewhere? Remove if possible.

**Review by rart — COMMENTED** [review]
Add prop changes to changelog

**Review by rart — CHANGES_REQUESTED** [review]
Comments from the previous review have not been address...

File: ui/app/src/CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,7 @@
     * `LegacyRecentActivityDashlet`
     * `LegacyRecentlyPublishedDashlet`
     * `LegacyUnpublishedDashlet`
+  * `WorkflowCancellationDialogUI`: prop `items` type changed from an async Resource to sync SandboxItem array.
```
Comment (jvega190) [inline ui/app/src/CHANGELOG.md:18]: The format for 4.2.0 changes is a bit different, so I put the component name only, under the components section

**Review by rart — APPROVED** [review]

**Comment by sumerjabri** [comment]
@jvega190 I'm unable to merge due to a conflict.

**Pull request merged**